core-web/libs/template-builder/src/lib/components/template-builder/store/template-builder.store.ts

```diff
@@ -0,0 +1,179 @@
+        });
+
+        return { items: itemsCopy };
+    });
+
+    /**
+     * @description This Method removes a row from the grid
+     *
+     * @memberof DotTemplateBuilderStore
+     */
+    readonly removeRow = this.updater(({ items }, id: string) => {
+        return { items: items.filter((item: DotGridStackWidget) => item.id !== id) };
+    });
+
+    /**
+     * @description This Method updates the row with the new data, as new styleclasses
+     *
+     * @memberof DotTemplateBuilderStore
+     */
+    readonly updateRow = this.updater(({ items }, payload: DotGridStackWidget) => {
+        const itemsCopy = structuredClone(items);
+        const rowIndex = itemsCopy.findIndex((item: DotGridStackWidget) => item.id === payload.id);
+        if (rowIndex > -1) itemsCopy[rowIndex] = { ...itemsCopy[rowIndex], ...payload };
+
+        return { items: itemsCopy };
+    });
+
+    // Columns Updaters
+
+    /**
+     * @description This Method adds a column to a row
+     *
+     * @memberof DotTemplateBuilderStore
+     */
+    readonly addColumn = this.updater(({ items }, payload: DotGridStackWidget) => {
+        const itemsCopy = structuredClone(items);
+        const rowIndex = itemsCopy.findIndex(
+            (item: DotGridStackWidget) => item.id === payload.parentId
+        );
+        if (rowIndex > -1)
+            if (itemsCopy[rowIndex].subGridOpts)
+                itemsCopy[rowIndex].subGridOpts.children.push(payload); // Add the node
+            else itemsCopy[rowIndex].subGridOpts = { children: [payload] };
+
+        return { items: itemsCopy };
+    });
+
+    /**
+     * @description This Method updates the position of the columns in Y axis
+     *
+     * @memberof DotTemplateBuilderStore
+     */
+    readonly moveColumnInYAxis = this.updater(
+        ({ items }, [nodeToDelete, nodeToAdd]: DotGridStackWidget[]) => {
+            const itemsCopy = structuredClone(items);
+
+            const deleteNodeRowIndex = itemsCopy.findIndex(
+                (item: DotGridStackWidget) => item.id === nodeToDelete.parentId
+            );
+            const addNodeRowIndex = itemsCopy.findIndex(
+                (item: DotGridStackWidget) => item.id === nodeToAdd.parentId
+            );
+
+            if (deleteNodeRowIndex > -1) {
+                itemsCopy[deleteNodeRowIndex].subGridOpts.children = itemsCopy[
+                    deleteNodeRowIndex
+                ].subGridOpts.children.filter(
+                    (child: DotGridStackWidget) => child.id !== nodeToDelete.id
+                ); // Filter the moved node
+            }
+
+            if (addNodeRowIndex > -1) {
+                if (itemsCopy[addNodeRowIndex].subGridOpts)
+                    itemsCopy[addNodeRowIndex].subGridOpts.children.push(nodeToAdd); // Add the node
+                else itemsCopy[addNodeRowIndex].subGridOpts = { children: [nodeToAdd] };
+            }
+
+            return {
+                items: itemsCopy
+            };
+        }
```

Comment:
We can use the `Utils.xxx` from gridstack for this complex transformations.

Reply:
made a whole refactor of this code to increase readability